### DISCUSSION TITLE
Null-conditional method call

### DISCUSSION THREAD

**Discussion by PavielKraskouski** [dotnet/csharplang#7407] [Language Ideas] [answered]
At the moment, C# has convenient null-conditional operators for member access and element acces operations.
I recently came across a situation where I need to get a value by calling a method if its argument is not `null`, otherwise copy the value from another variable. Now I can write it like this:
``` csharp
Vector3 baseColor = material.BaseColorMap == null ? material.BaseColor : SampleTexture(uv, material.BaseColorMap);
```
I suggest adding a null-conditional method call that returns `null` if the argument marked with `?` is `null`. So the previous code can be rewritten like this:
``` csharp
Vector3 baseColor = SampleTexture(uv, material.BaseColorMap?) ?? material.BaseColor;
```

**Comment by HaloFour** [comment, marked as answer]
See:

#3424
https://github.com/dotnet/roslyn/issues/5961
